### PR TITLE
add draft script to generate glo-30-hand service

### PR DESCRIPTION
**Pull request by asjohnston-asf**
This script inspects `s3://glo-30-hand/v1/2021/*HAND.tif` and creates:
- a [file geodatabase](https://pro.arcgis.com/en/pro-app/latest/help/data/geodatabases/manage-file-gdb/file-geodatabases.htm) (GLO30_HAND.gdb)
- a [mosaic dataset](https://pro.arcgis.com/en/pro-app/latest/help/data/imagery/mosaic-datasets.htm) within that geodatabase
- a [Cloud Raster Format](https://pro.arcgis.com/en/pro-app/latest/help/data/imagery/supported-raster-dataset-file-formats.htm) overview of the mosaic uploaded to `s3://hyp3-nasa-disasters/overviews/GLO30_HAND_Overview.crf`
- a Service Definition file (GLO30_HAND.sd) for [publishing the mosaic](https://enterprise.arcgis.com/en/server/latest/publish-services/windows/about-service-definition-files.htm) to ArcGIS Server

Individual arcpy commands are documented at https://pro.arcgis.com/en/pro-app/latest/tool-reference/data-management/an-overview-of-the-data-management-toolbox.htm , but you're better off googling "arcpy <command>" to jump straight to the reference page

### CODE REVIEW

File: image_services/glo_30_hand/make_hand_service.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    in_mosaic_dataset=mosaic_dataset,
+    raster_type='Raster Dataset',
+    input_path='/vsis3/glo-30-hand/v1/2021/',
+    filter='REGEX:.*Copernicus_DSM_COG_10_[NS][0-8]\d_00_[EW]\d\d\d_00_HAND.tif',
+)
+
+arcpy.management.CalculateFields(
+    in_table=mosaic_dataset,
+    fields=[
+        ['Tile', '!Name!.split("_")[4] + !Name!.split("_")[6]'],
```
Comment (jtherrmann) [inline image_services/glo_30_hand/make_hand_service.py:53]: What's this `!Name!` syntax in the string?

File: image_services/glo_30_hand/make_hand_service.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    update_missing_only='UPDATE_ALL',
+)
+
+with arcpy.EnvManager(cellSize=600):
+    arcpy.management.CopyRaster(
+        in_raster=mosaic_dataset,
+        out_rasterdataset=local_overview,
+    )
+
+subprocess.run(['aws', 's3', 'cp', local_overview, s3_overview.replace("/vsis3/", "s3://"), '--recursive'])
```
Comment (jtherrmann) [inline image_services/glo_30_hand/make_hand_service.py:128]: We could use `boto3` but I support the decision to use the `aws` cli if it makes this significantly simpler.

File: image_services/glo_30_hand/make_hand_service.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+arcpy.CreateImageSDDraft(
+    raster_or_mosaic_layer=mosaic_dataset,
+    out_sddraft=service_definition_draft,
+    service_name=dataset_name,
+    summary="Height Above Nearest Drainage (HAND) is a terrain model that normalizes topography to the relative " \
+            "heights along the drainage network and is used to describe the relative soil gravitational potentials " \
+            "or the local drainage potentials. Each pixel value represents the vertical distance to the nearest " \
+            "drainage. The HAND data provides near-worldwide land coverage at 30 meters and was produced from " \
+            "the 2021 release of the Copernicus GLO-30 Public DEM as distributed in the Registry of Open Data on " \
```
Comment (jtherrmann) [inline image_services/glo_30_hand/make_hand_service.py:161]: Should we double check the release version given the issue we identified for `hyp3-docs`? Or do we really mean the `2021` release this time?

File: image_services/glo_30_hand/make_hand_service.py (hunk 1 above)
Comment (asjohnston-asf) [reply image_services/glo_30_hand/make_hand_service.py:53]: The string on the right is an entire python expression to be evaluated. `!Name!` is a variable that contains the geotiff name without th3 `.tif`, e.g. `Copernicus_DSM_COG_10_N08_00_W065_00_HAND`

https://pro.arcgis.com/en/pro-app/latest/tool-reference/data-management/calculate-fields.htm
https://pro.arcgis.com/en/pro-app/latest/tool-reference/data-management/calculate-field-examples.htm

Comment (asjohnston-asf) [reply image_services/glo_30_hand/make_hand_service.py:53]: > For Python calculations, field names must be enclosed in exclamation points (for example, !fieldname!).

File: image_services/glo_30_hand/make_hand_service.py (hunk 3 above)
Comment (jtherrmann) [reply image_services/glo_30_hand/make_hand_service.py:161]: https://registry.opendata.aws/copernicus-dem/

>Data is provided as Cloud Optimized GeoTIFFs and comes from Copernicus DEM 2021 release.

File: image_services/glo_30_hand/make_hand_service.py (hunk 2 above)
Comment (jtherrmann) [reply image_services/glo_30_hand/make_hand_service.py:128]: Not blocking.

**Review by cirrusasf — APPROVED** [review]

**Pull request merged**